src/main/java/com/team766/framework/Context.java

```diff
@@ -132,15 +138,21 @@ private void threadFunction() {
 		} catch (Exception ex) {
 			ex.printStackTrace();
 			LoggerExceptionUtils.logException(ex);
+			Logger.get(Category.PROCEDURES).logRaw(Severity.WARNING, "Context " + getContextName() + " died");
 		} finally {
-			synchronized (m_threadSync) {
-				m_state = State.DONE;
-				m_threadSync.notifyAll();
-			}
 			for (Mechanism m : m_ownedMechanisms) {
 				// Don't use this.releaseOwnership here, because that would cause a
 				// ConcurrentModificationException since we're iterating over m_ownedMechanisms
-				m.releaseOwnership(this);
+				try {
+					m.releaseOwnership(this);
```

Comment:
it would, but that would require a new overload of `releaseOwnership` that takes an `Iterator` instead of a `Mechanism`, and that overload would only be used here. just calling `m_ownedMechanisms.clear();` below is simpler than removing mechanisms one-by-one as well